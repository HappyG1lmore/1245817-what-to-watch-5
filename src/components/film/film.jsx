import React, {PureComponent} from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import PropTypes from "prop-types";
import Header from "../header/header";
import Tabs from "../tabs/tabs";
import {getFilmInfo} from "../../store/api-action";

class Film extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {
      getFilmInformation,
      match: {
        params
      }
    } = this.props;
    const id = params.id;

    getFilmInformation(id);
  }

  render() {
    const {
      history,
      match,
      authorizationStatus
    } = this.props;
    const {params, url} = match;
    const {film} = this.props;

    return (
      film ?
        <React.Fragment>
          <section
            className="movie-card movie-card--full"
            style={{backgroundColor: `${film.backgroundColor}`}}
          >
            <div className="movie-card__hero">
              <div className="movie-card__bg">
                <img src={film.background} alt={film.title}/>
              </div>

              <h1 className="visually-hidden">WTW</h1>

              <Header className={`movie-card__head`}/>

              <div className="movie-card__wrap">
                <div className="movie-card__desc">
                  <h2 className="movie-card__title">{film.title}</h2>
                  <p className="movie-card__meta">
                    <span className="movie-card__genre">{film.genre}</span>
                    <span className="movie-card__year">{film.year}</span>
                  </p>

                  <div className="movie-card__buttons">
                    <button className="btn btn--play movie-card__button" type="button"
                      onClick={() => {
                        history.push(`/player/${params.id}`);
                      }}>
                      <svg viewBox="0 0 19 19" width="19" height="19">
                        <use xlinkHref="#play-s"/>
                      </svg>
                      <span>Play</span>
                    </button>
                    <button className="btn btn--list movie-card__button" type="button">
                      <svg viewBox="0 0 19 20" width="19" height="20">
                        <use xlinkHref="#add"/>
                      </svg>
                      <span>My list</span>
                    </button>
                    {
                      authorizationStatus === `AUTH` &&
                      <Link to={`${url}/review`} className="btn movie-card__button">Add review</Link>
                    }
                  </div>
                </div>
              </div>
            </div>

            <div className="movie-card__wrap movie-card__translate-top">
              <div className="movie-card__info">
                <div className="movie-card__poster movie-card__poster--big">
                  <img src={film.poster} alt={film.title} width="218"
                    height="327"/>
                </div>

                <Tabs film={film}/>

              </div>
            </div>
          </section>

          <div className="page-content">
            <section className="catalog catalog--like-this">
              <h2 className="catalog__title">More like this</h2>

              <div className="catalog__movies-list">
                <article className="small-movie-card catalog__movies-card">
                  <div className="small-movie-card__image">
                    <img src={`img/fantastic-beasts-the-crimes-of-grindelwald.jpg`}
                      alt="Fantastic Beasts: The Crimes of Grindelwald" width="280" height="175"/>
                  </div>
                  <h3 className="small-movie-card__title">
                    <a className="small-movie-card__link" href={`movie-page.html`}>
                      Fantastic Beasts: The Crimes of Grindelwald
                    </a>
                  </h3>
                </article>

                <article className="small-movie-card catalog__movies-card">
                  <div className="small-movie-card__image">
                    <img src={`img/bohemian-rhapsody.jpg`} alt="Bohemian Rhapsody" width="280" height="175"/>
                  </div>
                  <h3 className="small-movie-card__title">
                    <a className="small-movie-card__link" href="movie-page.html">Bohemian Rhapsody</a>
                  </h3>
                </article>

                <article className="small-movie-card catalog__movies-card">
                  <div className="small-movie-card__image">
                    <img src={`img/macbeth.jpg`} alt="Macbeth" width="280" height="175"/>
                  </div>
                  <h3 className="small-movie-card__title">
                    <a className="small-movie-card__link" href="movie-page.html">Macbeth</a>
                  </h3>
                </article>

                <article className="small-movie-card catalog__movies-card">
                  <div className="small-movie-card__image">
                    <img src={`img/aviator.jpg`} alt="Aviator" width="280" height="175"/>
                  </div>
                  <h3 className="small-movie-card__title">
                    <a className="small-movie-card__link" href="movie-page.html">Aviator</a>
                  </h3>
                </article>
              </div>
            </section>

            <footer className="page-footer">
              <div className="logo">
                <Link to={`/`} className="logo__link logo__link--light">
                  <span className="logo__letter logo__letter--1">W</span>
                  <span className="logo__letter logo__letter--2">T</span>
                  <span className="logo__letter logo__letter--3">W</span>
                </Link>
              </div>

              <div className="copyright">
                <p>Â© 2019 What to watch Ltd.</p>
              </div>
            </footer>
          </div>
        </React.Fragment>
        :
        <h2>Film with url: app{url} not found</h2>
    );
  }
}

Film.propTypes = {
  history: PropTypes.object,
  match: PropTypes.object,
  params: PropTypes.object,
  url: PropTypes.string,
  getFilmInformation: PropTypes.func,
  film: PropTypes.object,
  authorizationStatus: PropTypes.string
};

const mapStateToProps = (state) => {
  return {
    filmsList: state.films.filmsList,
    film: state.films.film,
    authorizationStatus: state.users.authorizationStatus
  };
};

const mapDispatchToProps = (dispath) => {
  return {
    getFilmInformation: (id) => dispath(getFilmInfo(id)),
  };
};

export {Film};
export default connect(mapStateToProps, mapDispatchToProps)(Film);

