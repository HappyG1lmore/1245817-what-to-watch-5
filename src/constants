
export const videoPlayerSize = {
  preview: {
    width: `280`,
    height: `175`
  }
};

export const timeoutTime = {
  oneSecond: 1000
};
