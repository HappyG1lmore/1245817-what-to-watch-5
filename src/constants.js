
export const ALL_GENRES = `All`;

export const FilmPreviewSize = {
  WIDTH: `280`,
  HEIGHT: `175`
};

export const HttpCode = {
  UNAUTHORIZED: 401
};

export const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`
};

export const PLAY_PREVIEW_TIMEOUT = 1000;
export const API_TIMEOUT = 5000;

export const APIRoute = {
  FILMS: `/films`,
  LOGIN: `/login`,
  COMMENTS: `/comments`,
};

export const TabType = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const MAX_STARRING = 4;

export const REVIEWS_COLUMN_COUNT = 3;

export const MINUTES_IN_HOUR = 60;

export const REVIEW_LENGTH = {
  MIN: 50,
  MAX: 400
};

export const MAX_AMOUNT_SIMILAR_FILM = 4;

