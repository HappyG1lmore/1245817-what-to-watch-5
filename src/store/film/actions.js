import {adaptFilmToClient} from "../../utils";

export const ActionType = {
  FETCH_FILM_INFO_SUCCESS: `FETCH_FILM_INFO_SUCCESS`,
  CLEAR_FILM_INFO: `CLEAR_FILM_INFO`,
  FETCH_FILM_PROMO_SUCCESS: `FETCH_FILM_PROMO_SUCCESS`
};

export const onFilmInfoFetchSuccess = (filmInfo) => {
  return {
    type: ActionType.FETCH_FILM_INFO_SUCCESS,
    payload: adaptFilmToClient(filmInfo)
  };
};

export const onFilmPromoFetchSuccess = (promoFilm) => {
  return {
    type: ActionType.FETCH_FILM_PROMO_SUCCESS,
    payload: adaptFilmToClient(promoFilm)
  };
};

export const clearFilmInfo = () => {
  return {
    type: ActionType.CLEAR_FILM_INFO,
  };
};

