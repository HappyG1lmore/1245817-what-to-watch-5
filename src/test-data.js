const filmsRaw = [{
  "name": `Bronson`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "background_color": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "rating": 3.6,
  "scores_count": 109661,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "run_time": 92,
  "genre": `Action`,
  "released": `2008`,
  "id": 1,
  "is_favorite": false,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Pulp Fiction`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
  "background_color": `#795433`,
  "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
  "rating": 1.5,
  "scores_count": 1635992,
  "director": `Quentin Tarantino`,
  "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  "run_time": 153,
  "genre": `Crime`,
  "released": `1994`,
  "id": 2,
  "is_favorite": false,
  "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "name": `Matrix`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
  "background_color": `#B9B27E`,
  "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
  "rating": 4.4,
  "scores_count": 1503092,
  "director": `Wachowski Brothers`,
  "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
  "run_time": 136,
  "genre": `Action`,
  "released": `1999`,
  "id": 4,
  "is_favorite": false,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}];

const films = [{
  "title": `Bronson`,
  "poster": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  "frame": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  "background": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  "backgroundColor": `#73B39A`,
  "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  "ratingText": `Normal`,
  "rating": 3.6,
  "ratings": `109661`,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "runtime": 92,
  "genre": `Action`,
  "year": `2008`,
  "id": 1,
  "isFavorite": false,
  "video": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "previewVideoLink": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "title": `Pulp Fiction`,
  "poster": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
  "frame": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
  "background": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
  "backgroundColor": `#795433`,
  "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
  "ratingText": `Bad`,
  "rating": 1.5,
  "ratings": `1635992`,
  "director": `Quentin Tarantino`,
  "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  "runtime": 153,
  "genre": `Crime`,
  "year": `1994`,
  "id": 2,
  "isFavorite": false,
  "video": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "previewVideoLink": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  "title": `Matrix`,
  "poster": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
  "frame": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
  "background": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
  "backgroundColor": `#B9B27E`,
  "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
  "ratingText": `Normal`,
  "rating": 4.4,
  "ratings": `1503092`,
  "director": `Wachowski Brothers`,
  "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
  "runtime": 136,
  "genre": `Action`,
  "year": `1999`,
  "id": 4,
  "isFavorite": false,
  "video": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "previewVideoLink": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}];

const film = {
  "title": `Pulp Fiction`,
  "poster": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
  "frame": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
  "background": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
  "backgroundColor": `#795433`,
  "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
  "rating": 1.5,
  "ratings": `1635992`,
  "director": `Quentin Tarantino`,
  "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  "runtime": 153,
  "genre": `Crime`,
  "year": `1994`,
  "id": 2,
  "isFavorite": false,
  "video": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  "previewVideoLink": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

const reviews = [{
  "id": 1,
  "user": {
    "id": 19,
    "name": `Christina`
  },
  "rating": 3.3,
  "comment": `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
  "date": `2020-10-09T13:38:44.769Z`
}, {
  "id": 2,
  "user": {
    "id": 25,
    "name": `Sophie`
  },
  "rating": 9,
  "comment": `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
  "date": `2020-11-04T13:38:44.769Z`
}, {
  "id": 3,
  "user": {
    "id": 7,
    "name": `Jack`
  },
  "rating": 9,
  "comment": `Unfortunately we don't have a reliable way to tell the true ratings of a movie.`,
  "date": `2020-10-09T13:38:44.769Z`
}];

const avatarUrl = `/avatar.jpg`;

const user = {
  "id": 1,
  "email": `Oliver.conner@gmail.com`,
  "name": `Oliver.conner`,
  "avatar_url": `img/1.png`
};

const noop = () => {};

const genres = [`Comedy`, `Drama`, `Thriller`];

export {films, genres, reviews, avatarUrl, user, film, filmsRaw, noop};
