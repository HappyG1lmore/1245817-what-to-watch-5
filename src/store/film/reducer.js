import {extend} from "../../utils";
import {ActionType} from "./actions";

const initialState = {
  isFilmInfoFetching: true,
  film: null,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.FETCH_FILM_INFO_SUCCESS:
      return extend(state, {
        isFilmInfoFetching: false,
        film: action.payload,
      });
    case ActionType.CLEAR_FILM_INFO:
      return extend(state, {
        isFilmInfoFetching: false,
        film: null,
      });
  }
  return state;
};

export {reducer};
