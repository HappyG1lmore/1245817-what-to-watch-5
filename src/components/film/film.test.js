import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {Film} from './film';
import {films, noop} from '../../test-data';
import {AuthorizationStatus} from '../../constants';


const renderer = new ShallowRenderer();

describe(`Film renders correctly`, () => {
  it(`When user is authorized`, () => {
    renderer.render(
        <Film
          history={{push: noop}}
          match={{params: {id: 3}}}
          url={`/films/3`}
          getFilmInformation={noop}
          film={films[2]}
          authorizationStatus={AuthorizationStatus.AUTH}
          clearFilm={noop}
          getCommentsList={noop}
          clearReviews={noop}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it(`When user not authorized`, () => {
    renderer.render(
        <Film
          history={{push: noop}}
          match={{params: {id: 3}}}
          url={`/films/3`}
          getFilmInformation={noop}
          film={films[2]}
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          clearFilm={noop}
          getCommentsList={noop}
          clearReviews={noop}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

});
