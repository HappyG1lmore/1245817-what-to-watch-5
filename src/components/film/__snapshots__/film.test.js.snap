// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film renders correctly When user is authorized 1`] = `
<React.Fragment>
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#B9B27E",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Matrix"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <Memo(Connect(Header))
        className="movie-card__head"
      />
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Matrix
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              1999
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <Memo(Connect(FavoriteButton))
              id={4}
              isFavorite={false}
            />
            <Link
              className="btn movie-card__button"
              to="undefined/review"
            >
              Add review
            </Link>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Matrix"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg"
            width="218"
          />
        </div>
        <ActiveTab
          film={
            Object {
              "background": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg",
              "backgroundColor": "#B9B27E",
              "description": "A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.",
              "director": "Wachowski Brothers",
              "frame": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg",
              "genre": "Action",
              "id": 4,
              "isFavorite": false,
              "poster": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": 4.4,
              "ratingText": "Normal",
              "ratings": "1503092",
              "runtime": 136,
              "starring": Array [
                "Keanu Reeves",
                "Laurence Fishburne",
                "Carrie-Anne Moss",
              ],
              "title": "Matrix",
              "video": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
              "year": "1999",
            }
          }
        />
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <Memo(Connect(LikeThis))
      genre="Action"
      id={4}
    />
    <Footer />
  </div>
</React.Fragment>
`;

exports[`Film renders correctly When user not authorized 1`] = `
<React.Fragment>
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#B9B27E",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Matrix"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <Memo(Connect(Header))
        className="movie-card__head"
      />
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Matrix
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              1999
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <Memo(Connect(FavoriteButton))
              id={4}
              isFavorite={false}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Matrix"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg"
            width="218"
          />
        </div>
        <ActiveTab
          film={
            Object {
              "background": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg",
              "backgroundColor": "#B9B27E",
              "description": "A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.",
              "director": "Wachowski Brothers",
              "frame": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg",
              "genre": "Action",
              "id": 4,
              "isFavorite": false,
              "poster": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": 4.4,
              "ratingText": "Normal",
              "ratings": "1503092",
              "runtime": 136,
              "starring": Array [
                "Keanu Reeves",
                "Laurence Fishburne",
                "Carrie-Anne Moss",
              ],
              "title": "Matrix",
              "video": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
              "year": "1999",
            }
          }
        />
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <Memo(Connect(LikeThis))
      genre="Action"
      id={4}
    />
    <Footer />
  </div>
</React.Fragment>
`;
